docs/_writing-tests/wdspec.md

```diff
@@ -0,0 +1,75 @@
+---
+layout: page
+title: wdspec tests
+order: -1
+---
+
+The term "wdspec" describes a type of test in WPT which verifies some aspect of
+[the WebDriver protocol](https://w3c.github.io/webdriver/). These tests are
+written in [the Python programming language](https://www.python.org/) and
+structured with [the pytest testing
+framework](https://docs.pytest.org/en/latest/).
+
+The test files are organized into subdirectories based on the WebDriver
+command under test. For example, tests for [the Close Window
+command](https://w3c.github.io/webdriver/#close-window) are located in then
+`close_window` directory.
+
+Similar to [testharness.js][], wdspec tests contain within them any number of
+"sub-tests." Sub-tests are defined as Python functions whose name begins with
+`test_`, e.g. `test_state_element`.
+
+## The `webdriver` client library
+
+web-platform-tests maintains a WebDriver client library called `webdriver`
+located in the `tools/webdriver/` directory. Like other client libraries, it
+makes it easier to write code which interfaces with a browser using the
+protocol.
+
+Many tests require some "set up" code--logic intended to bring the browser to a
+known state from which the expected behavior can be verified. The convenience
+methods in the `webdriver` library **should** be used to perform this task
```

Comment:
```suggestion
methods in the `webdriver` library **should** be used to perform this task,
```

Reply:
I'd agree if the sentence started with "Because," but when the dependent clause comes after the independent clause, [a comma isn't necessary](https://owl.purdue.edu/owl/general_writing/punctuation/commas/extended_rules_for_commas.html).